lib/cylc/config.py

```diff
@@ -1085,12 +1085,14 @@ def configure_queues(self):
         # Then reassign to other queues as requested.
         warnings = []
         requeued = []
-        for queue in queues:
-            if queue == 'default':
+        for key, queue in queues.copy().items():
+            # queues.copy() is essential here to allow items to be removed from
+            # the queues dict.
```

Comment:
A good discussion can be found here http://www.gossamer-threads.com/lists/python/python/1073911

We can in theory do `for key in list(queues.keys()):`, but since the value is used most of the time, I think it makes sense to use `.items()` instead.